library/src/main/java/com/google/maps/android/clustering/algo/GridBasedAlgorithm.java

```diff
@@ -63,6 +63,12 @@ public void removeItems(Collection<T> items) {
         mItems.removeAll(items);
     }
 
+    @Override
+    public void updateItem(T item) {
+        removeItem(item);
```

Comment:
It would be good to check for the existence of `item` within `mItems` before invoking `removeItem` and `addItem`. Otherwise, there could be unintended effects. For example, if the developer did not not override `equals` and `hashCode` on the concrete `ClusterItem`, `item` will likely not be removed from the `Set` and a duplicate marker would be added.

Same comment for other concrete instances of `Algorithm`.

Reply:
:+1: I went ahead and added returning boolean for all Algorithm add/remote/update operations to match Java Collection behavior in https://github.com/googlemaps/android-maps-utils/pull/627/commits/a1b1c17eac8d71ecbb0758bac6f1f514547c21f1, and added unit tests for them too. I could use another review on that just to make sure I didn't miss anything.